Flaky test: `test_contact_forces_on_incline` fails intermittently on CPU

## Summary

The test `test_mujoco_solver.TestMuJoCoContactForce.test_contact_forces_on_incline` is flaky and fails intermittently on CI (specifically on `macos-latest` runners), but cannot be reproduced locally.

## Test Failure Details

**Test:** `test_contact_forces_on_incline`  
**Location:** `newton/tests/test_mujoco_solver.py:4078`  
**Description:** Contact force on an incline must balance gravity (tests rotated contact frame).

### Error Message

```
AssertionError: 
Not equal to tolerance rtol=0.05, atol=0

Mismatched elements: 1 / 1 (100%)
Max absolute difference among violations: 78.48
Max relative difference among violations: 1.
 ACTUAL: array(0.)
 DESIRED: array(78.48)
```

**Expected behavior:** The Z-component of the contact force (`force[2]`) should be approximately 78.48 (BOX_MASS * GRAVITY)  
**Actual behavior:** The Z-component is 0.0, suggesting no contact forces are being computed

### Failing Location

```python
# Line 4084 in test_mujoco_solver.py
np.testing.assert_allclose(force[2], expected_fz, rtol=0.05)
```

## Environment

- **Runner:** `macos-latest` (GitHub Actions)
- **Python:** 3.12.10
- **Architecture:** arm64

## Known Failure Instances

Recent failures on the `main` branch:

1. **2026-03-25 21:01 UTC** - Run [#23563938100](https://github.com/newton-physics/newton/actions/runs/23563938100/job/68610864626)
   - Job: newton-unittests (macos-latest)
   - Status: FAILED

Additional recent failures on main branch (some may be related to this or other issues):
- 2026-03-25 20:09 - [Run #23561702599](https://github.com/newton-physics/newton/actions/runs/23561702599)
- 2026-03-25 13:40 - [Run #23544054897](https://github.com/newton-physics/newton/actions/runs/23544054897)
- 2026-03-24 13:41 - [Run #23492533501](https://github.com/newton-physics/newton/actions/runs/23492533501)
- 2026-03-23 17:07 - [Run #23450017515](https://github.com/newton-physics/newton/actions/runs/23450017515)

## Analysis

The test simulates a box on an inclined ramp and verifies that contact forces properly balance gravity. The test:

1. Settles the simulation for 300 steps
2. Averages contact forces over 80 additional steps
3. Expects the Z-component of force to match `BOX_MASS * GRAVITY ≈ 78.48`

The failure (force = 0.0) suggests one of the following:
- No active contacts are being detected (`nacon = 0`)
- Contact forces aren't being computed correctly
- Timing/settling issues specific to the CI environment
- Possible race condition or numerical precision issue on ARM macOS

## Reproducibility

**Local reproduction:** ❌ Cannot reproduce on similar local macOS ARM64 system  
**CI reproduction:** ✅ Intermittent failures on GitHub Actions macos-latest runners

This is a classic **flaky test** that passes most of the time but fails intermittently on CI.

## Proposed Solutions

1. **Increase settling time:** The test uses `settle=300` and `avg=80`. CI might need more settling steps.
2. **Add diagnostic logging:** Log `nacon`, actual force values, and contact state when the assertion fails.
3. **Relax tolerance:** Consider if `rtol=0.05` is too strict for this physics simulation.
4. **Add retry logic:** Mark the test with `@retry` or `@flaky` decorator for known flaky tests.
5. **Skip on CI:** As a last resort, skip this test on CI with `@unittest.skipIf` if it cannot be stabilized.
6. **Investigate MuJoCo determinism:** Check if MuJoCo behavior differs between local and CI environments.

## Additional Context

The test is part of the `TestMuJoCoContactForce` test class and is related to validating the MuJoCo solver's contact force computation in rotated contact frames (inclined surfaces).

--- a/newton/tests/test_mujoco_solver.py
+++ b/newton/tests/test_mujoco_solver.py
@@ -4459,7 +4459,6 @@ class TestMuJoCoContactForce(unittest.TestCase):
         model.request_contact_attributes("force")
         return model, ramp_shape
 
-    @unittest.skip("Flaky on CI, see GH-2239")
     def test_contact_forces_on_incline(self):
         """Contact force on an incline must balance gravity (tests rotated contact frame)."""
         incline_angle = 0.25  # rad (~14°); mu=1.0 > tan(0.25)≈0.26 → static

